### PR TITLE
Fixed stuff, implement github 

### PR DESCRIPTION
**Pull request by NotFlameDev**
- Fixed detect_calc got triggered on single negative number. E.g. `-1` or `--3`
- Implemented GitHub file editing
- Calc float support
- Fixed a bug on Calc not triggering with ( as the first character

### CODE REVIEW

File: src/cogs/ide/github.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+async def edit_file(repo: str, file_path: str, new_content: bytes, commit_message: str):
+    url = f"/repos/{repo}/contents/{file_path}"
+    content = base64.b64encode(new_content)
+    b64_content = content.decode("ascii")
+    r = await request("GET", url)
+    await request(
+        "PUT", url,
+        headers={
+            "Authorization": ...  # TODO: Get token from database
```
Comment (Pandabweer) [inline src/cogs/ide/github.py:21]: I don't like TODO comments inside code, please just finish the code, if you need something else, like a .env or config file please make that first.

File: src/cogs/ide/github.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import aiohttp
+import base64
+
+BASE_URL = "https://api.github.com"
+
+
+async def request(method, url, **kwargs):
+    async with aiohttp.ClientSession() as session:
```
Comment (Pandabweer) [inline src/cogs/ide/github.py:8]: Bot is getting http_session, please use that when [pull 102](https://github.com/CaedenPH/Jarvide/pull/102) is finished.

**Review by Pandabweer — CHANGES_REQUESTED** [review]
github.py needs to get changed or deleted.

File: src/cogs/ide/github.py (hunk 1 above)
Comment (NotFlameDev) [reply src/cogs/ide/github.py:21]: the token endpoint isn't ready yet, thus it's a TODO

**Review by Pandabweer — APPROVED** [review]

**Comment by NotFlameDev** [comment]
> whats this github.py?

GitHub simple API wrapper (not really a wrapper). The place to make requests to GitHub

**Comment by CaedenPH** [comment]
> > whats this github.py?
> 
> GitHub simple API wrapper (not really a wrapper). The place to make requests to GitHub

why do we need requests to github?

**Pull request merged**